packages/react/src/animation/TextRevealAnimation/useTextRevealAnimation.ts

```diff
@@ -1,7 +1,7 @@
 import {useRef, useEffect, useState} from 'react'
 
-export const useWrapLines = (text?: string) => {
-  const ref = useRef<HTMLHeadingElement>(null)
+export const useTextRevealAnimation = (text?: string) => {
```

Comment:
Minor thing, but I think the previous naming was more accurate here. This hook doesn't deal with any animation or revealing, it just allows you to wrap individual lines of text as you see fit. Maybe `useSplitByLine` or something similar could be another option.

Reply:
Yeah that's fair. I called it this for DevExp reasons, so we can find it easily while doing file searches. It can also be used as a general hook as we expand capabilities of the component, which is what we do for some other components too. I can see it the other way too though, as it does return some specific information.

Reply:
Cool, I can see the logic there.

My opinion would be that, since it's already co-located with the component (which tells us that it's related to the component), using a generic name instead of a descriptive one would make for a worse DevExp, as devs now have to read the implementation of the hook before they can understand what it does.

It also makes it more likely that additional, unrelated stuff will get piled into that hook in the future too. I take the point that it's convenient to have a single hook that contains all of a component's hook logic, but in the same way that we break big functions or components down into smaller, more focussed parts I'd tend towards doing the same with hooks. If a component ends up relying on a large number of those smaller hooks then that's the point I'd look to wrap up all the calls to those hooks into a single, component-named hook.

I'm happy with however you want to approach it though, just sharing my two cents 🙂

Reply:
👍 yeah I agree with those points. I've already renamed the hook to append Lines at the end as it was returning something rather than a void... Still has the component name prepended so best of both worlds.